packages/notifier/src/subscribe.js

```diff
@@ -0,0 +1,147 @@
+import { E, Far } from '@endo/far';
+
+import './types-ambient.js';
+
+const sink = () => {};
+
+/**
+ * Create a near iterable that corresponds to a potentially far one.
+ *
+ * @template T
+ * @param {ERef<AsyncIterable<T>>} itP
+ */
+export const subscribe = itP =>
+  Far('AsyncIterable', {
+    [Symbol.asyncIterator]: () => {
+      const it = E(itP)[Symbol.asyncIterator]();
+      return Far('AsyncIterator', {
+        next: async () => E(it).next(),
+      });
+    },
+  });
+
+/**
+ * Asyncronously iterates over the contents of a PublicationList as they appear.
+ * This iteration must drop parts of publication records that are no longer
+ * needed so they can be garbage collected.
+ *
+ * @template T
+ * @param {PublicationList<T>} pubList
+ * @returns {AsyncIterableIterator<T, T>}
+ */
+const makeEachIterator = pubList => {
+  // To understand the implementation, start with
+  // https://web.archive.org/web/20160404122250/http://wiki.ecmascript.org/doku.php?id=strawman:concurrency#infinite_queue
+  return Far('EachIterator', {
+    next: () => {
+      const resultP = E.get(pubList).head;
+      pubList = E.get(pubList).tail;
+      // We expect the tail to be the "cannot read past end" error at the end
+      // of the happy path.
+      // Since we are wrapping that error with eventual send, we sink the
+      // rejection here too so it doesn't become an invalid unhandled rejection
+      // later.
+      void E.when(pubList, sink, sink);
+      return resultP;
+    },
+    [Symbol.asyncIterator]: () => makeEachIterator(pubList),
```

Comment:
Per https://github.com/Agoric/agoric-sdk/pull/6866#discussion_r1107499408 . please rename to `fork` and update the README accordingly.